--- nginx.conf	2022-02-23 14:30:39.145780835 +0000
+++ nginx.conf_patched	2022-02-23 14:40:47.226983727 +0000
@@ -74,8 +74,16 @@

         root /usr/share/nginx/html;

-        location / {
-            proxy_pass http://${SERVICE}/;
+        location /naomi/ {
+            rewrite ^/naomi/(.*) /$1 break;
+            proxy_pass http://naomi_web:8888;
+            proxy_redirect http://naomi_web:8888/ /naomi/;
+        }
+
+        location /fertility/ {
+            rewrite ^/fertility/(.*) /$1 break;
+            proxy_pass http://fertility_orderly:8881;
+            proxy_redirect http://fertility_orderly:8881/ /fertility/;
         }
     }

